generated/nifgen/session.py

```diff
@@ -2696,15 +2713,22 @@ def _write_named_waveform_f64(self, waveform_name, data):
         Args:
             waveform_name (str): Specifies the name to associate with the allocated waveform.
 
-            data (list of float): Specifies the array of data to load into the waveform. The array must
+            data (array.array("d")): Specifies the array of data to load into the waveform. The array must
                 have at least as many elements as the value in **size**.
 
         '''
         vi_ctype = visatype.ViSession(self._vi)  # case S110
         channel_name_ctype = ctypes.create_string_buffer(self._repeated_capability.encode(self._encoding))  # case C010
         waveform_name_ctype = ctypes.create_string_buffer(waveform_name.encode(self._encoding))  # case C020
         size_ctype = visatype.ViInt32(0 if data is None else len(data))  # case S160
-        data_ctype = None if data is None else (visatype.ViReal64 * len(data))(*data)  # case B550
+        if data is not None:  # case B550
```

Comment:
this whole if/else structure could be a helper function

Reply:
It could but since this is generated code, the actual source is not duplicated, so what would it gain us? A few bytes in size for session.py?

Reply:
Improved readability in both the generated code and the generator.

Reply:
Moved into a helper function in _converters.py

Still do this in two lines, one for the array, one for the ctypes. I didn't make a single function that returned a tuple because then this case would not be able to use the common "var_ctype = definition" that you had commented about earlier.